explorer/src/components/instruction/bonfida-bot/types.ts

```diff
@@ -0,0 +1,429 @@
+  targetMint: string(),
+});
+
+export const decodeCreateOrder = (ix: TransactionInstruction): CreateOrder => {
+  const decoded = create(
+    decodeInstruction(ix.data, Instruction.CreateOrder),
+    CreateDecode
+  );
+
+  const createOrder: CreateOrder = {
+    programId: ix.programId,
+    signalProviderKey: ix.keys[0].pubkey,
+    market: ix.keys[1].pubkey,
+    payerPoolAssetKey: ix.keys[2].pubkey,
+    openOrdersKey: ix.keys[3].pubkey,
+    serumRequestQueue: ix.keys[5].pubkey,
+    serumEventQueue: ix.keys[4].pubkey,
+    serumMarketBids: ix.keys[6].pubkey,
+    serumMarketAsks: ix.keys[7].pubkey,
+    poolKey: ix.keys[8].pubkey,
+    coinVaultKey: ix.keys[9].pubkey,
+    pcVaultKey: ix.keys[10].pubkey,
+    splTokenProgramId: ix.keys[11].pubkey,
+    dexProgramKey: ix.keys[13].pubkey,
+    rentProgramId: ix.keys[12].pubkey,
+    srmReferrerKey: ix.keys[14]?.pubkey,
+    // Miss maxQuantity
+    //
+    poolSeed: decoded.poolSeed,
+    side: decoded.side,
+    limitPrice: decoded.limitPrice,
+    ratioOfPoolAssetsToTrade: decoded.ratioOfPoolAssetsToTrade,
+    orderType: decoded.orderType,
+    clientId: decoded.clientId,
+    coinLotSize: decoded.coinLotSize,
+    pcLotSize: decoded.pcLotSize,
+    targetMint: decoded.targetMint,
+  };
+
+  return createOrder;
+};
+
+export type CancelOrder = {
+  programId: PublicKey;
+  signalProviderKey: PublicKey;
+  market: PublicKey;
+  openOrdersKey: PublicKey;
+  serumEventQueue: PublicKey;
+  serumMarketBids: PublicKey;
+  serumMarketAsks: PublicKey;
+  poolKey: PublicKey;
+  dexProgramKey: PublicKey;
+  poolSeed: string;
+  side: Side;
+  orderId: number;
+};
+
+export const CancelOrderDecode = type({
+  poolSeed: string(),
+  side: Side,
+  orderId: number(),
+});
+
+export const decodeCancelOrder = (ix: TransactionInstruction): CancelOrder => {
+  const decoded = create(
+    decodeInstruction(ix.data, Instruction.CancelOrder),
+    CancelOrderDecode
+  );
+
+  const cancelOrder: CancelOrder = {
+    programId: ix.programId,
+    signalProviderKey: ix.keys[0].pubkey,
+    market: ix.keys[1].pubkey,
+    openOrdersKey: ix.keys[2].pubkey,
+    serumEventQueue: ix.keys[3].pubkey,
+    serumMarketBids: ix.keys[4].pubkey,
+    serumMarketAsks: ix.keys[5].pubkey,
+    poolKey: ix.keys[6].pubkey,
+    dexProgramKey: ix.keys[7].pubkey,
+    //
+    poolSeed: decoded.poolSeed,
+    side: decoded.side,
+    orderId: decoded.orderId,
+  };
+  return cancelOrder;
+};
+
+export type SettleFunds = {
+  programId: PublicKey;
+  market: PublicKey;
+  openOrdersKey: PublicKey;
+  poolKey: PublicKey;
+  poolMintKey: PublicKey;
+  coinVaultKey: PublicKey;
+  pcVaultKey: PublicKey;
+  coinPoolAssetKey: PublicKey;
+  pcPoolAssetKey: PublicKey;
+  vaultSignerKey: PublicKey;
+  splTokenProgramId: PublicKey;
+  dexProgramKey: PublicKey;
+  srmReferrerKey: PublicKey | null;
+  poolSeed: string;
+};
+
+export const SettleFundsDecode = type({
+  poolSeed: string(),
+  pcIndex: number(),
+  orderId: number(),
```

Comment:
`orderId` appears to be coming back `undefined` here, causing the Settle cards to render as a raw instruction:

![settlecardraw](https://user-images.githubusercontent.com/188792/116350071-27daea00-a7a6-11eb-8a4e-f447618908c7.png)